src/btree/bt_read.c

```diff
@@ -140,6 +141,242 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+                    cursor->prev(cursor);
+                    cursor->remove(cursor);
+                    cursor->next(cursor);
+                }
+
+                if (birthmark_record) {
+                    birthmarkp->instantiated = true;
+                    birthmark_record = false;
+                }
+                lastupd = NULL;
+            }
+            WT_ERR(__wt_buf_set(session, current_key, las_key.data, las_key.size));
+            break;
+        default:
+            WT_ERR(__wt_illegal_value(session, page->type));
+        }
+
+        if (ref->page_las->birthmarks_cnt > 0) {
+            for (; birthmark_cnt < ref->page_las->birthmarks_cnt; birthmark_cnt++) {
+                birthmarkp = ref->page_las->birthmarks + birthmark_cnt;
+                if (birthmarkp->instantiated)
+                    continue;
+
+                WT_ERR(__wt_compare(session, NULL, &las_key, &birthmarkp->key, &cmp));
+                if (cmp == 0) {
+                    WT_ERR(__create_birthmark_upd(session, birthmarkp, &upd));
+                    birthmark_record = true;
+                    break;
+                } else if (cmp < 0)
+                    /* LAS key is less than birthmark key, so no birthmark record for that key */
+                    break;
+                else
+                    /* LAS key is greater than birthmark key, check next set of birthmark records */
+                    continue;
+            }
+        }
+
+        if (upd == NULL) {
+            /* Allocate the WT_UPDATE structure. */
+            WT_ERR(
+              cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+            WT_ERR(__wt_update_alloc(session, &las_value, &upd, &incr, upd_type));
+            upd->txnid = las_txnid;
+            upd->durable_ts = durable_timestamp;
+            upd->start_ts = las_timestamp;
+            upd->prepare_state = prepare_state;
+        }
+
+        __wt_free(session, lastupd);
+        lastupd = upd;
+        upd = NULL;
+    }
+
+    if (lastupd != NULL) {
+        switch (page->type) {
+        case WT_PAGE_COL_FIX:
+        case WT_PAGE_COL_VAR:
+            WT_ERR(__col_instantiate(session, current_recno, ref, &cbt, lastupd));
+            break;
+        case WT_PAGE_ROW_LEAF:
+            WT_ERR(__row_instantiate(session, current_key, ref, &cbt, lastupd));
+            break;
+        default:
+            WT_ERR(__wt_illegal_value(session, page->type));
+        }
+
+        /* Remove the prepared record from LAS */
+        if (!birthmark_record && lastupd->prepare_state == WT_PREPARE_INPROGRESS) {
+            cursor->prev(cursor);
+            cursor->remove(cursor);
+        }
+
+        if (birthmark_record)
+            birthmarkp->instantiated = true;
+
+        lastupd = NULL;
+    }
+    __wt_readunlock(session, &cache->las_sweepwalk_lock);
+    locked = false;
+    WT_ERR_NOTFOUND_OK(ret);
```

Comment:
I think this needs to be moved just after the if loop, after line 330.

Reply:
done.